### DISCUSSION TITLE
Prisma db pull changes client to older version

### DISCUSSION THREAD

**Discussion by yazer79** [prisma/prisma#7189] [Q&A] [answered]
W10 / prisma 2.23.0
I have deleted the migration files to try to make a fresh Prisma install, and it made things worse
Trying Prisma migrate dev detects shift and wants to reset the db (i am adding to existing project, so I can't reset the db)
if I try db pull, the client is  changed to 2.20.1 (which i guess is the version where the migrate was made)
Stuck in the middle between wrong client, db shift, can't reset the db
Any suggestions?

**Comment by janpio** [comment, marked as answer]
How is your project set up? 
Are both `prisma` and `@prisma/client` project dependencies?
What commands _exactly_ are you using to execute the Prisma CLI?

**Reply by yazer79** [reply]
I have a mysql db
and I installed prisma via CLI / npm 
npm install prisma --save-dev
npm install @prisma/client
npx prisma introspect
prisma generate
prisma migrate dev
prisma db push / db pull

**Reply by janpio** [reply]
The command `prisma` with no `npx` in front of it indicate you also have a global `prisma` installation that is probably outdated. You can confirm that via `prisma -v` which will probably be different than `npx prisma -v`.

You should probably just get rid of the global install via `npm uninstall -g prisma` and use the local one for all commands via `npx prisma ...`.

**Reply by yazer79** [reply]
ok, thanks, i have uninstalled the global prisma, it was confusing me indeed as i didn't notice if i type prisma or npx prisma
But what about the db shift:
"Drift detected: Your database schema is not in sync with your migration history."
can i reset the migration without losing the db values?

**Reply by janpio** [reply]
That I can not tell you. Best open a new issue or discussion about this.

**Reply by yazer79** [reply]
ok, thanks a lot
